homeassistant/components/esphome/domain_data.py

```diff
@@ -21,30 +21,26 @@
 class DomainData:
     """Define a class that stores global esphome data in hass.data[DOMAIN]."""
 
-    _entry_datas: dict[str, RuntimeEntryData] = field(default_factory=dict)
     _stores: dict[str, ESPHomeStorage] = field(default_factory=dict)
     bluetooth_cache: ESPHomeBluetoothCache = field(
         default_factory=ESPHomeBluetoothCache
     )
 
-    def get_entry_data(self, entry: ConfigEntry) -> RuntimeEntryData:
+    def get_entry_data(self, entry: ESPHomeConfigEntry) -> RuntimeEntryData:
         """Return the runtime entry data associated with this config entry.
 
         Raises KeyError if the entry isn't loaded yet.
         """
-        return self._entry_datas[entry.entry_id]
+        return entry.runtime_data
 
-    def set_entry_data(self, entry: ConfigEntry, entry_data: RuntimeEntryData) -> None:
+    def set_entry_data(
+        self, entry: ESPHomeConfigEntry, entry_data: RuntimeEntryData
+    ) -> None:
         """Set the runtime entry data associated with this config entry."""
-        assert entry.entry_id not in self._entry_datas, "Entry data already set!"
-        self._entry_datas[entry.entry_id] = entry_data
```

Comment:
Would be nice to remove get and set in the future but too many things break and would make this PR to big to remove it at this point